File: svws-asd/src/main/java/de/svws_nrw/asd/validate/schueler/ValidatorSle00SchuelerLernabschnittsdatenEpJahre.java
package de.svws_nrw.asd.validate.schueler;

import java.util.function.Supplier;

import de.svws_nrw.asd.validate.Validator;
import de.svws_nrw.asd.validate.ValidatorKontext;
import de.svws_nrw.transpiler.annotations.AllowNull;
import jakarta.validation.constraints.NotNull;

/**
 * Prüft, ob die EP-Jahre zulässig, bzw. in der JSON-Datei vorhanden ist.
 */
public final class ValidatorSle00SchuelerLernabschnittsdatenEpJahre extends Validator {

	/** EP - Jahre */
	private final @NotNull Supplier<@AllowNull Long> _idEpJahre;
	private static final @NotNull String FEHLERTEXT = "EP-Jahr des Schülers: Das Feld 'EP-Jahr' darf nicht leer sein.";

	/**
	 * Erstellt einen neuen Validator zur Überprüfung der EP-Jahre.
	 *
	 * @param idEpJahre   EPJahreID
	 * @param kontext     der Kontext des Validators
	 */
	public ValidatorSle00SchuelerLernabschnittsdatenEpJahre(
			final @NotNull Supplier<@AllowNull Long> idEpJahre,
			final @NotNull ValidatorKontext kontext) {
		super(kontext);
		_idEpJahre = idEpJahre;

		_validatoren.add(new ValidatorSle01SchuelerLernabschnittsdatenEpJahre(getNotNullSupplierLong(idEpJahre), kontext));
	}

	@Override
	protected boolean pruefe() {
		// Bestimme epJahre.
		final @AllowNull Long idEpJahre = _idEpJahre.get();

			if (idEpJahre == null) {
				addFehler(0, FEHLERTEXT);
				return false;
			}
		return true;
	}
}

File: svws-webclient/core/src/asd/validate/schueler/ValidatorSle00SchuelerLernabschnittsdatenEpJahre.ts
import { ValidatorSle01SchuelerLernabschnittsdatenEpJahre } from '../../../asd/validate/schueler/ValidatorSle01SchuelerLernabschnittsdatenEpJahre';
import type { Supplier } from '../../../java/util/function/Supplier';
import { Class } from '../../../java/lang/Class';
import { ValidatorKontext } from '../../../asd/validate/ValidatorKontext';
import { Validator } from '../../../asd/validate/Validator';

export class ValidatorSle00SchuelerLernabschnittsdatenEpJahre extends Validator {

	/**
	 * EP - Jahre
	 */
	private readonly _idEpJahre: Supplier<number | null>;

	private static readonly FEHLERTEXT: string = "EP-Jahr des Schülers: Das Feld 'EP-Jahr' darf nicht leer sein.";


	/**
	 * Erstellt einen neuen Validator zur Überprüfung der EP-Jahre.
	 *
	 * @param idEpJahre   EPJahreID
	 * @param kontext     der Kontext des Validators
	 */
	public constructor(idEpJahre: Supplier<number | null>, kontext: ValidatorKontext) {
		super(kontext);
		this._idEpJahre = idEpJahre;
		this._validatoren.add(new ValidatorSle01SchuelerLernabschnittsdatenEpJahre(this.getNotNullSupplierLong(idEpJahre), kontext));
	}

	protected pruefe(): boolean {
		const idEpJahre: number | null = this._idEpJahre.get();
		if (idEpJahre === null) {
			this.addFehler(0, ValidatorSle00SchuelerLernabschnittsdatenEpJahre.FEHLERTEXT);
			return false;
		}
		return true;
	}

	transpilerCanonicalName(): string {
		return 'de.svws_nrw.asd.validate.schueler.ValidatorSle00SchuelerLernabschnittsdatenEpJahre';
	}

	isTranspiledInstanceOf(name: string): boolean {
		return ['de.svws_nrw.asd.validate.BasicValidator', 'de.svws_nrw.asd.validate.schueler.ValidatorSle00SchuelerLernabschnittsdatenEpJahre', 'de.svws_nrw.asd.validate.Validator'].includes(name);
	}

	public static readonly class = new Class<ValidatorSle00SchuelerLernabschnittsdatenEpJahre>('de.svws_nrw.asd.validate.schueler.ValidatorSle00SchuelerLernabschnittsdatenEpJahre');

}

export function cast_de_svws_nrw_asd_validate_schueler_ValidatorSle00SchuelerLernabschnittsdatenEpJahre(obj: unknown): ValidatorSle00SchuelerLernabschnittsdatenEpJahre {
	return obj as ValidatorSle00SchuelerLernabschnittsdatenEpJahre;
}
